tests/composer.json

```diff
@@ -65,8 +65,7 @@
             },
             "guzzle7": {
                 "require": {
-                    "guzzlehttp/guzzle": "~7.0",
-                    "guzzlehttp/promises": "^1.5.3"
+                    "guzzlehttp/guzzle": "~7.0"
```

Comment:
**suggestion:** As we are using `guzzlehttp/promises` in our code we might keep it in the `composer.json` to make this explicit. Otherwise we are relying on Guzzle bringing this dependency, which we do not control.

```suggestion
                    "guzzlehttp/guzzle": "~7.0",
                    "guzzlehttp/promises": "~2.0"
```